Tests/OpenAPIGeneratorReferenceTests/SnippetBasedReferenceTests.swift

```diff
@@ -4230,6 +4230,209 @@ final class SnippetBasedReferenceTests: XCTestCase {
         )
     }
 
+    func testRequestMultipartBodyReferencedSchemaRecursive() throws {
+        try self.assertRequestInTypesClientServerTranslation(
+            """
+            /foo:
+              post:
+                requestBody:
+                  required: true
+                  content:
+                    multipart/form-data:
+                      schema:
+                        $ref: '#/components/schemas/Node'
+                responses:
+                  default:
+                    description: Response
+            """,
+            """
+            schemas:
+              Node:
+                type: object
+                properties:
+                  log: 
+                    type: string
+                  parent:
+                    $ref: '#/components/schemas/Node'
+                required:
+                  - log
+            """,
```

Comment:
```suggestion
                        $ref: '#/components/schemas/NodeWrapper'
                responses:
                  default:
                    description: Response
            """,
            """
            schemas:
              NodeWrapper:
                type: object
                properties:
                  node:
                    $ref: '#/components/schemas/Node'
              Node:
                type: object
                properties:
                  parent:
                    $ref: '#/components/schemas/Node'
            """,
```

Can you try it like this? With multipart, the root object isn't really a struct at runtime, just a way to describe the individual multipart parts. But we still want to prove that it's possible to use recursive types in multipart _parts_.